frmts/hdf5/bagdataset.cpp

```diff
@@ -1441,6 +1471,708 @@ CPLErr BAGResampledBand::IReadBlock(int nBlockXOff, int nBlockYOff,
+    GDALRasterBand::SetDescription(nBand == 1 ? "elevation" : "uncertainty");
+    if (bInitializeMinMax)
+    {
+        InitializeMinMax();
+    }
+}
+
+/************************************************************************/
+/*                        ~BAGInterpolatedBand()                        */
+/************************************************************************/
+
+BAGInterpolatedBand::~BAGInterpolatedBand() = default;
+
+/************************************************************************/
+/*                           InitializeMinMax()                         */
+/************************************************************************/
+
+void BAGInterpolatedBand::InitializeMinMax()
+{
+    BAGDataset *poGDS = cpl::down_cast<BAGDataset *>(poDS);
+    if (nBand == 1 &&
+        GH5_FetchAttribute(poGDS->m_hVarresRefinements, "max_depth",
+                           m_dfMaximum) &&
+        GH5_FetchAttribute(poGDS->m_hVarresRefinements, "min_depth",
+                           m_dfMinimum))
+    {
+        m_bMinMaxSet = true;
+    }
+    else if (nBand == 2 &&
+             GH5_FetchAttribute(poGDS->m_hVarresRefinements, "max_uncrt",
+                                m_dfMaximum) &&
+             GH5_FetchAttribute(poGDS->m_hVarresRefinements, "min_uncrt",
+                                m_dfMinimum))
+    {
+        m_bMinMaxSet = true;
+    }
+}
+
+/************************************************************************/
+/*                             GetMinimum()                             */
+/************************************************************************/
+
+double BAGInterpolatedBand::GetMinimum(int *pbSuccess)
+
+{
+    if (m_bMinMaxSet)
+    {
+        if (pbSuccess)
+            *pbSuccess = TRUE;
+        return m_dfMinimum;
+    }
+
+    return GDALRasterBand::GetMinimum(pbSuccess);
+}
+
+/************************************************************************/
+/*                             GetMaximum()                             */
+/************************************************************************/
+
+double BAGInterpolatedBand::GetMaximum(int *pbSuccess)
+
+{
+    if (m_bMinMaxSet)
+    {
+        if (pbSuccess)
+            *pbSuccess = TRUE;
+        return m_dfMaximum;
+    }
+
+    return GDALRasterBand::GetMaximum(pbSuccess);
+}
+
+/************************************************************************/
+/*                     BarycentricInterpolation()                       */
+/************************************************************************/
+
+static bool BarycentricInterpolation(double dfX, double dfY,
+                                     const double adfX[3], const double adfY[3],
+                                     double &dfCoord0, double &dfCoord1,
+                                     double &dfCoord2)
+{
+    /* See https://en.wikipedia.org/wiki/Barycentric_coordinate_system */
+    double dfDenom = (adfY[1] - adfY[2]) * (adfX[0] - adfX[2]) +
```

Comment:
```suggestion
    const double dfDenom = (adfY[1] - adfY[2]) * (adfX[0] - adfX[2]) +
```

and all subsequent doubles

Reply:
done